samples/bluetooth/central_uart/src/main.c

```diff
@@ -355,12 +348,124 @@ static int scan_init(void)
 	return err;
 }
 
+static void gatt_discover(struct bt_conn *conn)
+{
+	if (conn == default_conn) {
+		int err = bt_gatt_dm_start(conn,
+					   BT_UUID_NUS_SERVICE,
+					   &discovery_cb,
+					   &gatt_nus_c);
+		if (err) {
+			printk("could not start the discovery procedure, error "
+			       "code: %d\n", err);
+		}
+	}
+}
+
+static void auth_passkey_display(struct bt_conn *conn, unsigned int passkey)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Passkey for %s: %06u\n", addr, passkey);
+}
+
+static void auth_passkey_confirm(struct bt_conn *conn, unsigned int passkey)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	auth_conn = conn;
```

Comment:
Need to use `bt_conn_ref` here, and `bt_conn_unref` when the pointer is assigned to NULL again.

Reply:
Ok